### DISCUSSION TITLE
Idiomatic way to use pystring with multithread and return it

### DISCUSSION THREAD

**Discussion by Crispy13** [PyO3/pyo3#4103] [Questions]
I would like to do the following:

1. Get Vec<Bound<,pystring>> from python
2. Do parallel job with the vector. I should convert the elements into sendable object first.
3. In par iter collect into Vec<(Py₩<PyString₩>, i32)>
4. Return it

Could you tell me idiomatic way ?

I tried myself but something got weird.
Another pure python multiprocessing function was stuck after the pyo3 function.

My trial was making a wrapper struct having py₩<pystring₩> and pybackedstr. 
I used str slice from pybackedstr for rust function in par iter and return py₩<pystring₩>.

**Comment by mejrs** [comment]
It's hard to recommend an approach without understanding *why* you want to do this, but you should probably get a `Vec<&str>` and send that to your parallel job.
